commercetools-models/src/test/java/io/sphere/sdk/carts/commands/CartCreateCommandIntegrationTest.java

```diff
@@ -199,12 +199,11 @@ public void createByJson() {
     }
 
     @Test
-    public void createWithDiscount() throws Exception {
-        DiscountCodeFixtures.withPersistentDiscountCode(client(), discountCode -> {
-            final CartDraft cartDraft = CartDraft.of(EUR).withCountry(DE).withDiscountCodes(singletonList(discountCode.getCode()));
-            final Cart cart = client().executeBlocking(CartCreateCommand.of(cartDraft));
-
-            assertThat(cart.getDiscountCodes().get(0).getDiscountCode().getId()).isEqualTo( discountCode.getId());
+    public void createCartWithDiscount() {
+        CartFixtures.withCartAndPersistentDiscountCode(client(), (cart, discountCode) -> {
```

Comment:
This changes the test behavior. Before was tested that you can apply the discount while creating the cart. Now you update the cart to add the discount.

Reply:
ok, I'm gonna fix that :)

Reply:
I fixed it in another branch. Found the culprit